brave/brave-core
# fix multiple sync issues

This is a bundle of fixes for sync and is included together in one PR for convenience since the spec was fully updated and wouldn't make sense to have them separated. 

**Design spec:** https://share.goabstract.com/ef5f3ea9-3aec-472e-81d0-1ad72d2a70e5

## 1) Update spec of Sync enabled view

### Fix **Clicking anywhere on the same line as 'Bookmarks' toggles Bookmark Sync on/off**
Close https://github.com/brave/brave-browser/issues/2574

### Test plan

1. Sync two devices
2. Once Sync is enabled, clicking the bookmark text shouldn't trigger the toggle on/off.

## 2) Update spec of sync disabled view

### Fix **Text in the first screen (disabled content) feels extremely cramped and uneven**
Close https://github.com/brave/brave-browser/issues/2675

### Test plan

1. This is set as QA/No as it is a design change and was addressed with the latest design spec (see the top of this comment for the link).

## 3) Update spec of Sync Devices/QR code/Sync Words modals

### Fix **Disable Phone/Computer buttons until the device name is queried**
Close https://github.com/brave/brave-browser/issues/2724

### Fix **Padding is too small between `Word Code` and `QR Code` in `View Sync Code` modal**
Close https://github.com/brave/brave-browser/issues/2591

### Test plan

**Note:** This was addressed by the new design spec by removing the device name from the device type modal and the remake of the QR code and sync words.

**First fix** (https://github.com/brave/brave-browser/issues/2724)

1. Create a new chain
2. In the Device Type modal (one with computer/mobile icons) ensure there isn't a device name in the title

**Second fix** (https://github.com/brave/brave-browser/issues/2591)

1. With two devices connected, click "view sync code" and check that `word code` and `QR code` are different modals.

## 4) Add a loading state for adddin, removing and resetting Sync

### Fix **Sync needs a 'waiter' message when reset/confirm/remove**
Close https://github.com/brave/brave-browser/issues/2567

### Fix **Looking for device button is missing when QR code is shown**
Close https://github.com/brave/brave-browser/issues/2801

### Fix **Prevent the user from firing device removal multiple times**
Close https://github.com/brave/brave-browser/issues/2629

### Test plan

**First fix** (https://github.com/brave/brave-browser/issues/2567)

1. Clean profile
2. Click "start a new sync chain"
3. Click either mobile or computer
4. Check that there is a loading spinner looking for devices
5. Copy the sync code
6. Cancel the sync chain and go back to the first screen
7. Click "I have a code" and confirm your code
8. See the loading spinner running until the new screen is rendered

**Second fix** (https://github.com/brave/brave-browser/issues/2629)

1. Have 3 devices in a sync chain
2. Once devices are loaded, remove one device by clicking the X button
confirm removal
3. All buttons should be disabled
4. A loading spinner should show while the operation is running
modal should self-close once the operation is completed

## 5) Add extra dialog to prevent the user from aborting a new sync chain

### Fix **when the user starts a new sync chain the modal needs to disable cancelation**
Close https://github.com/brave/brave-browser/issues/2564

### Test plan

1. Clear profile
2. Click "start a new sync chain"
3. Click "tablet or phone"
4. Try to cancel the modal
5. A new modal appears asking if you are sure about closing it

Repeat the same process but this time clicking "computer".

## 6) Properly set the state for `view code words` and `scan words`

### Fix **Children of Sync device type modal should use the same modal**
Close https://github.com/brave/brave-browser/issues/2677

### Test plan

1. Clear profile
2. Click "Start a New Sync Chain"
3. Click "Tablet or Phone"
4. Keep toggling between "View Sync Code" and "View QR Code"
5. Trying to close it closes them all at once

Repeat the same process with a sync chain set up by clicking "Add New Device" in the devices table screen.

## 7) Auto-close code modals when a new device is found

### Fix **`Cancel` and `OK` yield the same action in `add device`**
Close https://github.com/brave/brave-browser/issues/2550

### Fix **Looking for devices seems to be a dead UI on sync page question**
Close https://github.com/brave/brave-browser/issues/2528

### Test plan

**First fix** (https://github.com/brave/brave-browser/issues/2550)
1. Have a Sync chain enabled
2. Click "Add a New Device"
3. Should show a disabled button with a label of "looking for devices"

**Second fix** (https://github.com/brave/brave-browser/issues/2528)
1. Clear profile
2. Create a new sync chain either by mobile or computer options
3. Sync another device with the Sync Code Words/QR Code
4. Check that the second device is synced
5. Wait a few moments and ensure the modal in the first device auto-closes once a new device is found

## 8) Rename resetSyncSetupError to clearSyncSetupError

### Fix **Sync: rename resetSyncSetupError to clearSyncSetupError**
Close https://github.com/brave/brave-browser/issues/2779

### Test plan

n/a
